modules/core/src/main/scala/lucuma/odb/api/model/TargetModel.scala

```diff
@@ -331,72 +402,192 @@ object TargetModel extends TargetOptics {
   }
 
 }
 
 trait TargetOptics { self: TargetModel.type =>
 
-  val id: Lens[TargetModel, Target.Id] =
-    Lens[TargetModel, Target.Id](_.id)(a => b => b.copy(id = a))
-
-  val existence: Lens[TargetModel, Existence] =
-    Lens[TargetModel, Existence](_.existence)(a => b => b.copy(existence = a))
+  val name: Lens[Target, NonEmptyString] =
+    Target.name
 
-  val lucumaTarget: Lens[TargetModel, Target] =
-    Lens[TargetModel, Target](_.target)(a => b => b.copy(target = a))
+  val nonsiderealTarget: Optional[Target, Target] =
+    Optional.filter[Target](_.track.isLeft)
 
-  val name: Lens[TargetModel, NonEmptyString] =
-    lucumaTarget.andThen(Target.name)
+  val siderealTarget: Optional[Target, Target] =
+    Optional.filter[Target](_.track.isRight)
 
-  private val gemTargetEphemerisKey: Optional[Target, EphemerisKey] =
+  val ephemerisKey: Optional[Target, EphemerisKey] =
     Target.track.andThen(monocle.std.either.stdLeft[EphemerisKey, SiderealTracking])
 
-  val ephemerisKey: Optional[TargetModel, EphemerisKey] =
-    lucumaTarget.andThen(gemTargetEphemerisKey)
-
-  private val gemTargetSiderealTracking: Optional[Target, SiderealTracking] =
+  val siderealTracking: Optional[Target, SiderealTracking] =
     Target.track.andThen(monocle.std.either.stdRight[EphemerisKey, SiderealTracking])
 
-  val siderealTracking: Optional[TargetModel, SiderealTracking] =
-    lucumaTarget.andThen(gemTargetSiderealTracking)
-
-  val catalogId: Optional[TargetModel, Option[CatalogId]] =
+  val catalogId: Optional[Target, Option[CatalogId]] =
     siderealTracking.andThen(SiderealTracking.catalogId)
 
-  val coordinates: Optional[TargetModel, Coordinates] =
+  val coordinates: Optional[Target, Coordinates] =
     siderealTracking.andThen(SiderealTracking.baseCoordinates)
 
-  val ra: Optional[TargetModel, RightAscension] =
+  val ra: Optional[Target, RightAscension] =
     coordinates.andThen(Coordinates.rightAscension)
 
-  val dec: Optional[TargetModel, Declination] =
+  val dec: Optional[Target, Declination] =
     coordinates.andThen(Coordinates.declination)
 
-  val epoch: Optional[TargetModel, Epoch] =
+  val epoch: Optional[Target, Epoch] =
     siderealTracking.andThen(SiderealTracking.epoch)
 
-  val properMotion: Optional[TargetModel, Option[ProperMotion]] =
+  val properMotion: Optional[Target, Option[ProperMotion]] =
     siderealTracking.andThen(SiderealTracking.properMotion)
 
-  val radialVelocity: Optional[TargetModel, Option[RadialVelocity]] =
+  val radialVelocity: Optional[Target, Option[RadialVelocity]] =
     siderealTracking.andThen(SiderealTracking.radialVelocity)
 
-  val parallax: Optional[TargetModel, Option[Parallax]] =
+  val parallax: Optional[Target, Option[Parallax]] =
     siderealTracking.andThen(SiderealTracking.parallax)
 
-  val magnitudes: Lens[TargetModel, SortedMap[MagnitudeBand, Magnitude]] =
-    lucumaTarget.andThen(Target.magnitudes)
+  val magnitudes: Lens[Target, SortedMap[MagnitudeBand, Magnitude]] =
```

Comment:
Is this a circular reference?

Reply:
Actually now it's just a reference to `Target.magnitudes`. `TargetModel` the case class went away because without the id it offered nothing that `lucuma.core.model.Target` didn't.